src/data_science/data_science.py

```diff
@@ -713,6 +778,102 @@ def get_verkehrsleistung_by_config_id(self, config_id: UUID) -> pd.DataFrame:
+            lambda row: row["leave_tick"] - row["enter_tick"], axis=1
+        )
+        grouped_df = block_section_times_df.groupby(["run_id", "train_type"]).apply(
+            lambda data: pd.Series(
+                {
+                    "verkehrsmenge": data["block_section_length"].sum(),
+                }
+            )
+        )
+        grouped_df.reset_index(inplace=True)
+        all_df = grouped_df.groupby("run_id").apply(
+            lambda data: pd.Series(
+                {
+                    "train_type": "all",
+                    "verkehrsmenge": data["verkehrsmenge"].sum(),
+                }
+            )
+        )
+        all_df.reset_index(inplace=True)
+        grouped_df = pd.concat([grouped_df, all_df], axis=0)
+        grouped_df = grouped_df.groupby(["train_type"]).agg(
+            {"verkehrsmenge": "mean"}
+        )
+        grouped_df.reset_index(inplace=True)
+        return grouped_df
+
+    def get_average_verkehrsleistung_by_config_id(self, config_id: UUID) -> pd.DataFrame:
+        """Returns the average verkehrsleistung by a given config id
+        :param config_id: config id
+        :return: verkehrsleistung dataframe"""
+        df_list = []
+        for run_id in Run.select().where(Run.simulation_configuration == config_id):
+            block_section_times_df = (
+                self.log_collector.get_block_section_times_all_trains(run_id)
+            )
+            block_section_times_df["train_type"] = block_section_times_df["train_id"].apply(
+                lambda train_id: train_id.split("_")[2])
+            block_section_times_df["run_id"] = run_id.id
+            df_list.append(block_section_times_df)
+        block_section_times_df = pd.concat(df_list, axis=0)
+        block_section_times_df.dropna(inplace=True)
+        block_section_times_df["time"] = block_section_times_df.apply(
+            lambda row: row["leave_tick"] - row["enter_tick"], axis=1
+        )
+        grouped_df = block_section_times_df.groupby(["run_id", "train_type"]).apply(
+            lambda data: pd.Series(
+                {
+                    "enter_tick": 0,
+                    "leave_tick": data["leave_tick"].max(),
+                    "block_section_length": data["block_section_length"].sum(),
+                }
+            )
+        )
+        grouped_df.reset_index(inplace=True)
+        all_df = grouped_df.groupby("run_id").apply(
+            lambda data: pd.Series(
+                {
+                    "train_type": "all",
+                    "enter_tick": 0,
+                    "leave_tick": data["leave_tick"].max(),
+                    "block_section_length": data["block_section_length"].sum(),
+                }
+            )
+        )
+        all_df.reset_index(inplace=True)
+        grouped_df = pd.concat([grouped_df, all_df], axis=0)
+
+
+        grouped_df["verkehrsleistung"] = grouped_df.apply(
+            lambda row: row["block_section_length"] * 3600 / row["leave_tick"],
```

Comment:
"3600" equals the conversion from seconds to hours, correct?

Reply:
Yes, from km/s to km/h